### ISSUE TITLE
Discord Connector issues with functions not registered

### ISSUE THREAD

**Issue by domacx3** [maddinat0r/samp-discord-connector#113]
https://prnt.sc/pxlwsz

libssl.so.1.0.0 is installed and it still returns these errors and doesn't run the plugin at all. I am really mad right now because I haven't worked a lot with Linux and I've spent too much time figuring out how to install it in the first place. Can anyone please help me asap? thanks

**Comment by maddinat0r** [comment]
Make sure you install the 32bit libssl library. You can also check if the library is found by running `ldd path/to/discord-connector.so` (should print the libssl path if it's found).

**Comment by domacx3** [comment]
> Make sure you install the 32bit libssl library. You can also check if the library is found by running `ldd path/to/discord-connector.so` (should print the libssl path if it's found).

We have it installed but it's still not working.

**Comment by maddinat0r** [comment]
I've released v0.3.1, which is (except for log-core) fully statically linked, so you shouldn't have any errors regarding missing libraries. Please try this new version and report back if it works for you.

**Comment by domacx3** [comment]
It says it requires some plugin named _GLIBCXX_3.4.22_ but my friend said he'll install it and then I'll let you know how it goes once he does it. :)

Thank you very much for the new release though!

**Comment by domacx3** [comment]
He can't seem to get it working, it keeps returning this:
`Failed (/usr/lib/i386-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.22' not found (required by plugins/discord-connector.so))`

**Comment by maddinat0r** [comment]
Are you absolutely sure that you're using v0.3.1?

**Comment by domacx3** [comment]
> Are you absolutely sure that you're using v0.3.1?

Yes, sir! I downloaded the v0.3.1 for Linux, uploaded it, overwritten the original files, restarted the server and it just returns the message I mentioned above.

`Failed (/usr/lib/i386-linux-gnu/libstdc++.so.6: version GLIBCXX_3.4.22' not found (required by plugins/discord-connector.so))`

**Comment by maddinat0r** [comment]
There's definitely something wrong with the runtime linking for the Linux build. I'll look into it.

**Comment by maddinat0r** [comment]
Could you please try this build and report back if it works for you? https://ci.appveyor.com/project/maddinat0r/samp-discord-connector/branch/master/job/9n98gpesmlionb4o/artifacts

**Comment by domacx3** [comment]
I will check it out now and let you know how it goes.

**Comment by domacx3** [comment]
It still doesn't work and returns this message:
`Failed (/lib/i386-linux-gnu/libc.so.6: version 'GLIBC_2.25' not found (required by plugins/discord-connector.so))`

My friend told me to send you this image too, he says these are the versions we have installed on our servers: https://cdn.discordapp.com/attachments/648956644184817675/651847986636390428/Screenshot_163.png

**Comment by maddinat0r** [comment]
The changes I made did work, however, there's a new problem now. Due to OpenSSL being linked statically, it may rely on libc functions, that are provided by the system at compile-time (i.e. the AppVeyor CI environment), but not at runtime (your server environment). That's why you're now getting the `GLIBC_2.25 not found` error, instead of the `GLIBCXX_3.4.22 not found` one. I'll see what I can do to circumvent that, but it looks like just linking to OpenSSL dynamically is the most straightforward solution here.

**Comment by domacx3** [comment]
Hey, sorry for not replying in a while, my PC was broken.

Anyways, my friend re-installed our VPS and w/e and now it's working LMAO. I don't know how, but it just started working like a charm..

I just have one quick question - how do you use OnMessageCreate? What I mean is, how do you add commands in the latest version?

**Comment by maddinat0r** [comment]
I've worked around the libc issue by using an older OpenSSL version and released that as v0.3.2. I'll close this issue as the original problem should be fully resolved now.
<br>

> I just have one quick question - how do you use OnMessageCreate? What I mean is, how do you add commands in the latest version?

`OnMessageCreate` has a `DCC_Message:message` parameter, and with that you can use all message functions provided by the plugin:
```pawn
native DCC_GetMessageId(DCC_Message:message, dest[DCC_ID_SIZE], max_size = DCC_ID_SIZE);
native DCC_GetMessageChannel(DCC_Message:message, &DCC_Channel:channel);
native DCC_GetMessageAuthor(DCC_Message:message, &DCC_User:author);
native DCC_GetMessageContent(DCC_Message:message, dest[], max_size = sizeof dest);
native DCC_IsMessageTts(DCC_Message:message, &bool:is_tts);
native DCC_IsMessageMentioningEveryone(DCC_Message:message, &bool:mentions_everyone);
native DCC_GetMessageUserMentionCount(DCC_Message:message, &mentioned_user_count);
native DCC_GetMessageUserMention(DCC_Message:message, offset, &DCC_User:mentioned_user);
native DCC_GetMessageRoleMentionCount(DCC_Message:message, &mentioned_role_count);
native DCC_GetMessageRoleMention(DCC_Message:message, offset, &DCC_Role:mentioned_role);
```
So what you'd do for a command processor, is to get the actual message through `DCC_GetMessageContent`, and check if it starts with your expected command name. From here on it's the same as with every command (like in `OnPlayerCommandText` for example).
